_optimisations/matlab/vectorisation.md

```diff
@@ -0,0 +1,75 @@
+---
+level: 1
+published: true
+authors: Fred Sonnenwald, Robert Chisholm
+
+name: Vectorisation
+language: [MATLAB]
+subcategory: [Core]
+tags: [array, maths, loops, conditional]
+---
+
+MATLAB is optimised to take advantage of vectorisation for mathematical operations between arrays, whereby the processor executes one instruction across multiple variables simultaneously. Vectorisation can perform mathematical operations many times faster than a for loop and even take advantage of conditional logic.
+
+<!--more-->
+
+## Example Code
+
+To take advantage of [vectorisation within MATLAB](https://uk.mathworks.com/help/matlab/matlab_prog/vectorization.html), operations should be performed at an array-level, rather than by looping over the individual elements.
+
+The below example multiplies every element of an array by 2 using a for loop.
+
+```matlab
+% Allocate an array of 1 million elements
+vals = 1:1000000;
+% Multiply all elements by 2
+for ii=1:length(vals)
+    vals(ii) = vals(ii) * 2;
+end
+```
+
+To use vectorisation, it can be rewritten as a single line of code
+
+```matlab
+% Allocate an array of 1 million elements
+vals = 1:1000000;
+% Multiply all elements by 2
+vals = vals .* 2;
```

Comment:
```suggestion
vals = vals * 2;
```

Suggestion to undo Fred's scalar multiply suggestion i hastily merged before changing my mind.